lib/lightning_web/live/workflow_live.ex

```diff
@@ -224,6 +278,97 @@ defmodule LightningWeb.WorkflowLive do
      )}
   end
 
+  @impl true
+  def handle_event("delete-job", %{"id" => id}, socket) do
+    if socket.assigns.can_edit_job do
+      job = Jobs.get_job!(id)
+
+      case Jobs.delete_job(job) do
+        {:ok, _} ->
+          LightningWeb.Endpoint.broadcast!(
+            "project_space:#{socket.assigns.project.id}",
+            "update",
+            %{workflow_id: job.workflow_id}
+          )
+
+          {:noreply,
+           socket
+           |> put_flash(:info, "Job deleted successfully")
+           |> push_patch(
+             to:
+               Routes.project_workflow_path(
+                 socket,
+                 :show,
+                 socket.assigns.project.id,
+                 socket.assigns.current_workflow.id
+               )
+           )}
+
+        {:error, _} ->
+          {:noreply,
+           socket
+           |> put_flash(
+             :error,
+             "Unable to delete this job because it has downstream jobs"
+           )}
+      end
+    else
+      {:noreply,
+       socket
+       |> put_flash(:error, "You are not authorized to perform this action.")
+       |> push_patch(
+         to:
+           Routes.project_workflow_path(
+             socket,
+             :show,
+             socket.assigns.project.id,
+             socket.assigns.current_workflow.id
+           )
+       )}
+    end
+  end
+
+  @impl true
+  def handle_event("copied-to-clipboard", _, socket) do
+    {:noreply,
+     socket
+     |> put_flash(:info, "Copied webhook URL to clipboard")}
+  end
+
+  @impl true
+  def handle_event(
+        "create-job",
+        _,
+        %{assigns: %{can_create_job: true}} = socket
+      ) do
+    {:noreply,
+     socket
+     |> push_patch(
+       to:
+         Routes.project_workflow_path(
+           socket,
+           :new_job,
+           socket.assigns.project.id,
+           socket.assigns.current_workflow.id
+         )
+     )}
+  end
+
+  def handle_event(
+        "create-job",
+        _,
+        %{assigns: %{can_create_job: false}} = socket
+      ) do
+    {:noreply,
+     socket
+     |> put_flash(:error, "You are not authorized to perform this action.")}
+  end
```

Comment:
I think this may be overusing pattern matching here imo, I think it's easier to understand the two paths this code may take by using an if statement in one function than having two functions over several lines.